libraries/chain/transaction.cpp

```diff
@@ -70,7 +70,7 @@ fc::microseconds transaction::get_signature_keys( const vector<signature_type>&
          auto now = fc::time_point::now();
          EOS_ASSERT( now < deadline, tx_cpu_usage_exceeded, "transaction signature verification executed for too long ${time}us",
                      ("time", now - start)("now", now)("deadline", deadline)("start", start) );
-         auto[ itr, successful_insertion ] = recovered_pub_keys.emplace( sig, digest );
+         auto[ itr, successful_insertion ] = recovered_pub_keys.emplace( sig, digest, fc::check_canonical_t::no );
```

Comment:
Do you need to check `allow_non_canonical_signatures` is true?

Reply:
No, we don't check here anymore, we do the check when the block is applied, and we know whether the protocol feature is activated.